model/modeldecoder/metricset.go

```diff
@@ -86,74 +84,28 @@ type metricsetDecoder struct {
 	*utility.ManualDecoder
 }
 
-func (md *metricsetDecoder) decodeSamples(input interface{}, hasShortFieldNames bool) []*metricset.Sample {
-	if input == nil {
-		md.Err = errors.New("no samples for metric event")
-		return nil
-	}
-	raw, ok := input.(map[string]interface{})
-	if !ok {
-		md.Err = errors.New("invalid type for samples in metric event")
-		return nil
-	}
-
+func (md *metricsetDecoder) decodeSamples(input map[string]interface{}, hasShortFieldNames bool, out *[]metricset.Sample) {
 	fieldName := field.Mapper(hasShortFieldNames)
 	inverseFieldName := field.InverseMapper(hasShortFieldNames)
 
-	samples := make([]*metricset.Sample, len(raw))
-	i := 0
-	value := fieldName("value")
-	for name, s := range raw {
-		if s == nil {
-			continue
-		}
-		sampleMap, ok := s.(map[string]interface{})
-		if !ok {
-			md.Err = fmt.Errorf("invalid sample: %s: %s", name, s)
-			return nil
-		}
-
-		samples[i] = &metricset.Sample{
-			Name:  inverseFieldName(name),
-			Value: md.Float64(sampleMap, value),
-		}
-		if md.Err != nil {
-			return nil
-		}
-		i++
+	valueFieldName := fieldName("value")
+	for name, s := range input {
+		sampleObj, _ := s.(map[string]interface{})
```

Comment:
Why are you removing the error check here?

Reply:
It's guaranteed to be that type, by virtue of having passed JSON Schema validation. Since I moved schema validation into the exported Decode methods, you can no longer get here without passing validation.